WcaOnRails/app/models/upload_json.rb

```diff
@@ -0,0 +1,136 @@
+              value2: individual_results[1],
+              value3: individual_results[2],
+              value4: individual_results[3],
+              value5: individual_results[4],
+            }
+            new_res = InboxResult.new(new_result_attributes)
+            # Using this way of setting the attribute saves one SELECT per result
+            # to validate the competition presence.
+            # (a lot of time considering all the results to import!)
+            new_res.competition = competition
+            results_to_import << new_res
+          end
+
+          # Import scrambles for round
+          round["groups"].each do |group|
+            ["scrambles", "extraScrambles"].each do |scramble_type|
+              group[scramble_type]&.each_with_index do |scramble, index|
+                new_scramble_attributes = {
+                  competitionId: competition_id,
+                  eventId: event["eventId"],
+                  roundTypeId: round["roundId"],
+                  groupId: group["group"],
+                  isExtra: scramble_type == "extraScrambles",
+                  scrambleNum: index+1,
+                  scramble: scramble,
+                }
+                scrambles_to_import << Scramble.new(new_scramble_attributes)
+              end
+            end
+          end
+        end
+      end
+      begin
+        ActiveRecord::Base.transaction do
+          InboxPerson.where(competitionId: competition_id).delete_all
+          InboxResult.where(competitionId: competition_id).delete_all
+          Scramble.where(competitionId: competition_id).delete_all
+          InboxPerson.import!(persons_to_import)
+          Scramble.import!(scrambles_to_import)
+          InboxResult.import!(results_to_import)
+        end
+        true
+      rescue ActiveRecord::RecordNotUnique
+        errors.add(:results_file, "Duplicate personId in JSON.")
+        false
+      rescue ActiveRecord::RecordInvalid => invalid
+        object = invalid.record
+        if object.class == Scramble
+          errors.add(:results_file, "Scramble in '#{Round.name_from_attributes_id(object.eventId, object.roundTypeId)}' is invalid (#{invalid.message}), please fix it!")
+        elsif object.class == InboxPerson
+          errors.add(:results_file, "Person #{object.name} is invalid (#{invalid.message}), please fix it!")
+        elsif object.class == InboxResult
+          errors.add(:results_file, "Result for person #{object.personId} in '#{Round.name_from_attributes_id(object.eventId, object.roundTypeId)}' is invalid (#{invalid.message}), please fix it!")
+        else
+          # FIXME: that's actually not supposed to happen, as the only 3 types of records we create are above
+          errors.add(:results_file, "An invalid record prevented the results from being created: #{invalid.message}")
+        end
+        false
+      end
+    else
+      false
+    end
+  end
+
+  # FIXME: what is this used for?
```

Comment:
You added this one, any reason for that?

Reply:
It comes from https://github.com/thewca/worldcubeassociation.org/blob/master/WcaOnRails/app/models/results_submission.rb#L24.
I'll just remove it, and wait for Jeremy to reply on your other comment.